docs/guide/using-your-ide.md

```diff
@@ -0,0 +1,243 @@
+It is simply a matter of setting up the IDE correctly to fulfill the Webots controller build rules.
+
+Documenting every IDE for each OS is a huge and unmaintainable task.
+This is why only some of them are documented.
+They could serve as a reference for others.
+
+The build rules are:
+
+- The target executable file should have the same name as the controller directory, and follow strictly this path: `$WEBOTS_PROJECT/controllers/$(CONTROLLER_NAME)/$(CONTROLLER_NAME)$(EXE_EXTENSION)`.
+- The executable should be linked with the Webots controller library (and eventually include the corresponding path):
+
+%tab-component "language"
+
+%tab "C"
+
+```
+libraries:
+- $(WEBOTS_HOME)/lib/$(SL_PREFIX)Controller$(SL_SUFFIX)
+
+includes:
+- $(WEBOTS_HOME)/include/controller/c
+```
+
+%tab-end
+
+%tab "C++"
+
+```
+libraries:
+- $(WEBOTS_HOME)/lib/$(SL_PREFIX)Controller$(SL_SUFFIX)
+- $(WEBOTS_HOME)/lib/$(SL_PREFIX)CppController$(SL_SUFFIX)
+
+includes:
+- $(WEBOTS_HOME)/include/controller/cpp
+```
+
+%tab-end
+
+%tab "Java"
+
+```
+libraries:
+- $(WEBOTS_HOME)/lib/java/Controller.jar
+```
+
+%tab-end
+
+%tab "Python"
+
+```
+N/A because Python is interpreted.
+```
+
+%tab-end
+
+%end
+
+---
+
+*Legend*
+
+- `WEBOTS_HOME` is the path to Webots.
+- `WEBOTS_PROJECT` is the path to your Webots project.
+- `CONTROLLER_NAME` is the name of your controller.
+- `EXE_EXTENSION` is the executable file suffix.
+    - For C or C++: `.exe` on Windows, and nothing else.
+    - For Python: `.py`.
+    - For Java: `.class` or `.jar`.
+- `SL_PREFIX` is the prefix of a shared library: `lib` on Linux or macOS, and nothing on Windows.
+- `SL_SUFFIX` is the suffix of a shared library: `.so` on Linux, `.dylib` on macOS and `.dll` on Windows.
+- `PYTHON_VERSION` is your Python version, but concatenated (`27`, `37`, etc.).
+
+---
+
+### Visual Studio
+
+On Windows, [Visual Studio](https://visualstudio.microsoft.com) can be used to create a C or C++ controller.
+
+A `Visual Studio` project can be simply created using the "New Robot Controller..." wizard.
+To do so, you should simply select the **Wizards / New Robot Controller...** menu item, choose the C or C++ language, and select a Visual Studio project.
+The target project can be open in `Visual Studio`.
+
+### CMake
+
+[CMake](https://cmake.org) is a cross-platform free and open-source software tool for managing the build process of software using a compiler-independent method.
+Using [its generators](https://cmake.org/cmake/help/v3.0/manual/cmake-generators.7.html), it generates native build environments, such as `XCode`, `CodeBlocks`, `Sublime Text 2` or `Eclipse` projects.
+The actual build is processed in these environments.
+For the `Visual Studio` target, prefer the [solution above](#visual-studio).
+
+As a template, you could copy the following `CMakeLists.txt` file to your controller directory.
+This template is only a sample, it may be adapted depending on your CMake target.
+
+```cmake
```

Comment:
Instead of this long piece of configuration file, wouldn't it be simpler to add a wizard for cmake? That would also be more consistent with other IDEs, such as Visual Studio.